File: Perl/algorithmic-crush.pl
#!/usr/bin/perl
use strict;
use warnings;

# Enter your code here. Read input from STDIN. Print output to STDOUT
my $nm=<STDIN>;
chomp $nm;
my ($n,$m)=split / /,$nm;
#my @val=split (//,0 x ($n+1));
my @val=split (//,0 x $n);
for (my $r=0;$r<$m;$r++){
   my $tmp=<STDIN>;
   chomp ($tmp);
   my ($a, $b, $k)=split / /,$tmp;
   $val[$a-1]+=$k;
   if ($b<$n){
       $val[$b]-=$k;
   }
   print join(" ",@val)."\n";
}
my $max= $val[0];
my $sum= $val[0];
foreach (1..$n-1){
  $sum+=$val[$_];
  if ($max<$sum){
      $max=$sum;
  }
}
print "$max";

File: C/algorithmic-crush.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;
 
typedef long long ll;
 
int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
    ll n, m, l, r, k, sum=0, max=0;
    cin >> n >> m;
    vector<ll> arr(n+1);
    for(int i=1; i <= n; i++)
        arr[i] = 0;
    for(int i=1; i <= m; i++) {
        cin >> l >> r >> k;
        arr[l] += k;
        if(r <= n) arr[r+1] -= k;
    }
    max = arr[1];
    sum = arr[1];
    for(int i=2; i <= n; i++) {
        sum+=arr[i];
        if(max < sum) max = sum;
    }
    cout << max;
    return 0;
}
